example/dj/apps/app/tests/data_processors.py

```diff
@@ -380,3 +377,85 @@ def test_create_issue_via_user_one_to_one_renamed_fail(self, number, issue_data,
         assert_http_bad_request(resp)
         assert_in('leading_issue_renamed', self.deserialize(resp).get('messages', {}).get('errors'))
+
+    @data_provider('get_users_data')
+    def test_should_raise_bad_request_with_invalid_filename(self, number, data):
+        data['contract'] = {
+            'filename': 'contract',
+            'content': base64.b64encode(
+                ('Contract of %s code: šří+áýšé' % data['email']).encode('utf-8')
+            ).decode('utf-8')
+        }
+        resp = self.post(self.USER_API_URL, data=data)
+        assert_http_bad_request(resp)
+        errors = self.deserialize(resp).get('messages', {}).get('errors')
+        assert_in('contract', errors)
+        assert_equal(errors['contract'],
+                     'Content type cannot be evaluated from the filename please send it or change the filename')
+
+    @data_provider('get_users_data')
+    def test_should_raise_bad_request_if_file_content_is_not_in_base64(self, number, data):
+        data['contract'] = {
+            'content_type': 'text/plain',
+            'filename': 'contract.txt',
+            'content': 'abc',
+        }
+        resp = self.post(self.USER_API_URL, data=data)
+        assert_http_bad_request(resp)
+        errors = self.deserialize(resp).get('messages', {}).get('errors')
+        assert_in('contract', errors)
+        assert_equal(errors['contract']['content'], ugettext('File content must be in base64 format'))
+
+    @data_provider('get_users_data')
+    def test_should_raise_bad_request_if_url_is_not_valid(self, number, data):
+        data['contract'] = {
+            'filename': 'contract.txt',
+        }
+        resp = self.get_file_url_response(data, serve_url='hs://foo.bar/testfile.pdf')
+        assert_http_bad_request(resp)
+        errors = self.deserialize(resp).get('messages', {}).get('errors')
+        assert_in('contract', errors)
+        assert_equal(errors['contract']['url'], ugettext('Enter a valid URL.'))
+
+    @data_provider('get_users_data')
+    def test_should_raise_bad_request_if_required_items_are_not_provided(self, number, data):
+        REQUIRED_ITEMS = {'filename', 'content'}
+        REQUIRED_URL_ITEMS = {'filename', 'url'}
+        data['contract'] = {}
+        resp = self.post(self.USER_API_URL, data=data)
+        assert_http_bad_request(resp)
+        errors = self.deserialize(resp).get('messages', {}).get('errors')
+        assert_in('contract', errors)
+        msg = ugettext('File data item must contains {} or {}').format(
+                        ', '.join(REQUIRED_ITEMS), ', '.join(REQUIRED_URL_ITEMS)
+                    )
+        assert_equal(errors['contract'], msg)
+
+    @data_provider('get_users_data')
+    def test_should_raise_bad_request_if_file_is_unreachable(self, number, data):
+        data['contract'] = {
+            'filename': 'contract',
+        }
+        resp = self.get_file_url_response(data, call_url='http://foo.wbar/ttestfile.pdf')
```

Comment:
připravil bych responses tak aby vracel 404

Reply:
myslim, ze zde se ani response pripravovat nemusi aby to hodilo bad request - file unreachable

Reply:
zkouší to pak volat internet což nechceme